staging/src/k8s.io/client-go/util/workqueue/queue.go

```diff
@@ -181,6 +183,15 @@ func (q *Type) Add(item interface{}) {
 	q.cond.Signal()
 }
 
+// IsQueued returns a bool to indicate whether `item` is actively queued.
+// It returns true if the item is in the dirty set which means it is
+// queued or about to be queued once processing completes.
```

Comment:
"queued *or about to be queued*" in the doc of `IsQueued` is really confusing :-)

Reply:
it's also confusing that `IsQueued()` != "present in `q.queue`"

Reply:
would it be clearer to connect this to the `Add` function, like `IsAdded`?

once we start stacking interfaces like the DelayingQueue with AddAfter, it gets really confusing about which `Is___` function relates to which `Add_` function

Reply:
If we name it IsAdded() then we can call the delayed version IsDelayed() when an item is waiting as opposed to Added() "what I have been calling in the active queue".  Is that better?

Reply:
Not sure.  If you only have `isAdded`, it could also mean (confuse people) that it also checks for items added with `AddAfter`/`AddWithOptions` that have a delay. Potential `IsAddedAfter` is also complicated because `AddAfter` can add directly to the queue when duration is 0.

On the other hand, it would be nice to be consistent in the naming of querying methods on DelayingInterface where we are going to have `IsWaiting`, `DoneWaiting` and `LenWaiting`.

Can we consider `IsQueued` again?

And have `IsWaiting` and `IsQueuedOrWaiting` on the DelayingInterface? Then the distinction when asking for the state should be clearer.

> "queued or about to be queued" in the doc of IsQueued is really confusing :-)

Can we simplify the doc and just say "queued" ?

> it's also confusing that IsQueued() != "present in q.queue"

That's just an internal implementation and dirty needs to be considered when looking at it, but it shouldn't matter when describing/reading the interface.

> once we start stacking interfaces like the DelayingQueue with AddAfter, it gets really confusing about which Is___ function relates to which Add_ function

`AddAfter`/`AddWithOptions` have more complex behavior, so it cannot be easily inferred from the name alone.

Reply:
How about IsActivelyQueued()?  Or IsQueuedProcessable() - coupled with IsQueuedWaiting()?
I updated the confusing documentation on the last commit - does that look better?  Let me know what you think is best and I'll change it once we all agree the best name for the function.